### PR TITLE
ENT-3682 Added Snowflake based completion report

### PR DESCRIPTION
**Pull request by irfanuddinahmad**
**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-analytics-data-api doesn't install it
    - `test-master.in` if edx-analytics-data-api pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the edx-analytics-data-api requirements installed or if you checked out edx-enterprise-data into the src directory used by edx-analytics-data-api, you can run this command through an edx-analytics-data-api shell.
        - It would be `./manage.py makemigrations` in the shell.
- [ ] [Version](https://github.com/edx/edx-enterprise-data/blob/master/enterprise/__init__.py) bumped
- [ ] [Changelog](https://github.com/edx/edx-enterprise-data/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/edx-enterprise-data/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [Travis](https://travis-ci.org/github/edx/edx-enterprise-data), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise-data/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-analytics-data-api](https://github.com/edx/edx-analytics-data-api) to upgrade dependencies (including edx-enterprise-data)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-analytics-data-api will look for the latest version in PyPi.
    - Note: the edx-enterprise-data constraint in edx-analytics-data-api **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

File: enterprise_reporting/reporter.py [hunk 1]
```diff
@@ -58,6 +59,78 @@ class EnterpriseReportSender(object):
...
+                bee.lms_user_id,
+                bee.lms_courserun_key
+            FROM
+                enterprise.ent_base_enterprise_enrollment as bee
+            LEFT JOIN
+                enterprise.ent_base_enterprise_customer as bec
+            ON
+                bee.enterprise_customer_uuid = bec.enterprise_customer_uuid
+            WHERE
+                bec.enterprise_customer_uuid = '{enterprise_id}'
```
Comment (HammadAhmadWaqas) [inline enterprise_reporting/reporter.py]: I think we are not using `ent_base_enterprise_customer` we can skip JOIN to  `ent_base_enterprise_customer`

```
SELECT
                bee.enterprise_customer_uuid,
                bee.lms_user_id,
                bee.lms_courserun_key
            FROM
                enterprise.ent_base_enterprise_enrollment as bee
            WHERE
                bee.enterprise_customer_uuid = '{enterprise_id}'
```

File: enterprise_reporting/send_enterprise_reports.py [hunk 2]
```diff
@@ -17,7 +17,7 @@
 
 logging.basicConfig(level=logging.INFO)
 LOGGER = logging.getLogger(__name__)
-DATA_TYPES = ['progress', 'progress_v2', 'catalog']
+DATA_TYPES = ['progress', 'progress_v2', 'catalog', 'grade']
```
Comment (HammadAhmadWaqas) [inline enterprise_reporting/send_enterprise_reports.py]: I think you want to add 'completion' here as well.

File: venv/.Python [hunk 3]
```diff
@@ -0,0 +1 @@
+/System/Library/Frameworks/Python.framework/Versions/2.7/Python
```
Comment (HammadAhmadWaqas) [inline venv/.Python]: What is the purpose of this file?

File: venv/include/python2.7 [hunk 4]
```diff
@@ -0,0 +1 @@
+/System/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7
```
Comment (HammadAhmadWaqas) [inline venv/include/python2.7]: this as well

**Review by moconnell1453 — APPROVED** [review]

**Review by sameenfatima78 — APPROVED** [review]

**Review by HammadAhmadWaqas — APPROVED** [review]
please squash commits before merging it.

**Pull request merged**